src/components/List/ItemListMapper.tsx

```diff
@@ -0,0 +1,44 @@
+import { List } from "@/pages/list";
+import { sortByAlphabet, sortByCategory } from "./SortBySwitches";
+
+type ItemListMapperProps = {
+  itemList: List;
+  sortBy: string;
+};
+
+export function ItemListMapper(props: ItemListMapperProps) {
+  let sectionName = "";
+  let sorted: List = [];
+
+  switch (props.sortBy) {
+    case "date":
+      sorted = props.itemList;
+      break;
+    case "category":
+      sorted = sortByCategory(props.itemList);
+      break;
+    case "alphabetical":
+      sorted = sortByAlphabet(props.itemList);
+      break;
+  }
+  return (
+    <>
+      {sorted.map((product) => {
+        let nameSection = false;
+
+        if (props.sortBy === "category") {
+          if (product.category !== sectionName) {
+            sectionName = product.category;
+            nameSection = true;
+          }
+        }
+        return (
+          <div className=" py-1">
+            {nameSection ? <p>{sectionName + ":"}</p> : <p></p>}
```

Comment:
is it necessary to have an empty `<p></p>` here? 
Could we use `nameSection && <p>{sectionName + ":"}</p>`

Reply:
I changed it accordingly
Sry for assigning Dan i didn't read the timeline